File: Student.cpp
#include "Student.h"
#include <iostream>
using namespace std;

// Constructor
Student::Student(string id, string fName, string lName, string dept) {
    studentID  = id;
    firstName  = fName;
    lastName   = lName;
    department = dept;
    cout << "[Student Created]: " << firstName << " " << lastName << "\n";
}

// Destructor
Student::~Student() {
    cout << "[Student Removed]: " << firstName << " " << lastName << "\n";
}

// Getters
string Student::getStudentID()  { return studentID; }
string Student::getFirstName()  { return firstName; }
string Student::getLastName()   { return lastName; }
string Student::getDepartment() { return department; }

// Setters
void Student::setStudentID(string id)    { studentID  = id; }
void Student::setFirstName(string name)  { firstName  = name; }
void Student::setLastName(string name)   { lastName   = name; }
void Student::setDepartment(string dept) { department = dept; }

// Methods
void Student::displayInfo() {
    cout << "  Student ID : " << studentID << "\n";
    cout << "  Name       : " << firstName << " " << lastName << "\n";
    cout << "  Department : " << department << "\n";
}

File: Student.ts
import { StudentDef } from "./StudentDef";

export class Student extends StudentDef {
  constructor(studentId?: string, firstName?: string, lastName?: string, department?: string) {
    super();
    this.setStudentId(studentId   || "");
    this.setFirstName(firstName   || "");
    this.setLastName(lastName     || "");
    this.setDepartment(department || "");
  }

  getStudentId(): string  { return this.studentId; }
  getFirstName(): string  { return this.firstName; }
  getLastName(): string   { return this.lastName; }
  getDepartment(): string { return this.department; }

  setStudentId(studentId: string): void   { this.studentId  = studentId; }
  setFirstName(firstName: string): void   { this.firstName  = firstName; }
  setLastName(lastName: string): void     { this.lastName   = lastName; }
  setDepartment(department: string): void { this.department = department; }
}